### PR TITLE
Disable type-checking for Vue.js

### LINKED ISSUE #3876: Vue file analysis should not crash for parsing issues
When analyzing Vue files, we use `AnalysisWithWatchProgram`, the execution by the ESLlint bridge does not have any fallbacks. Therefore, when a parsing error occurs, the process crashes, raising an exception at the plugin level.

**We should not fail in case of parsing errors.**

The file provoking the error can be found here: https://github.com/directus/directus/blob/cdf8d94ec1e9d6e948660c890e3e7d1876fc6890/api/src/cli/commands/init/index.ts

This issue has been on Peach since at least March 20th, a more recent occurrence can be found here: https://cirrus-ci.com/task/4514938685751296?logs=analyze#L181

Here is a snippet of the stack trace:

```
ERROR: Failed to analyze file [api/src/cli/commands/init/index.ts]: File '@directus/tsconfig/node18-esm.json' not found.
ERROR: Failure during analysis, Node.js command to start eslint-bridge was: node /tmp/cirrus-ci-build/directus/workspace/.scannerwork/.sonartmp/eslint-bridge-bundle/package/bin/server 39639 127.0.0.1 /tmp/cirrus-ci-build/directus/workspace/.scannerwork true false /tmp/cirrus-ci-build/directus/workspace/.scannerwork/.sonartmp/eslint-bridge-bundle/package/custom-rules15585289546640949514/package
java.lang.IllegalStateException: Failed to analyze file api/src/cli/commands/init/index.ts
	at org.sonar.plugins.javascript.eslint.AnalysisProcessor.processParsingError(AnalysisProcessor.java:146)
	at org.sonar.plugins.javascript.eslint.AnalysisProcessor.processResponse(AnalysisProcessor.java:91)
	at org.sonar.plugins.javascript.eslint.AnalysisWithWatchProgram.analyze(AnalysisWithWatchProgram.java:156)
	at org.sonar.plugins.javascript.eslint.AnalysisWithWatchProgram.analyzeTsConfig(AnalysisWithWatchProgram.java:127)
	at org.sonar.plugins.javascript.eslint.AnalysisWithWatchProgram.analyzeFiles(AnalysisWithWatchProgram.java:84)
	at org.sonar.plugins.javascript.eslint.TypeScriptSensor.analyzeFiles(TypeScriptSensor.java:98)
	at org.sonar.plugins.javascript.eslint.AbstractEslintSensor.execute(AbstractEslintSensor.java:73)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:64)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:88)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:61)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:79)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:61)
	at org.sonar.scanner.scan.SpringModuleScanContainer.doAfterStart(SpringModuleScanContainer.java:82)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:188)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:167)
	at org.sonar.scanner.scan.SpringProjectScanContainer.scan(SpringProjectScanContainer.java:403)
	at org.sonar.scanner.scan.SpringProjectScanContainer.scanRecursively(SpringProjectScanContainer.java:399)
	at org.sonar.scanner.scan.SpringProjectScanContainer.doAfterStart(SpringProjectScanContainer.java:368)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:188)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:167)
	at org.sonar.scanner.bootstrap.SpringGlobalContainer.doAfterStart(SpringGlobalContainer.java:137)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:188)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:167)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:126)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:81)
	at org.sonarsource.scanner.cli.Main.main(Main.java:62)
```

### PR DESCRIPTION
**Pull request by vdiez**
Fixes #3876 #4352

Related: #3992

### CODE REVIEW

File: packages/jsts/tests/analysis/analyzer.test.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ describe('analyzeJSTS', () => {
     );
   });
 
-  it('should analyze Vue.js code with type checks', async () => {
+  it('Vue.js analysis does not support type checking', async () => {
```
Comment (zglicz) [inline packages/jsts/tests/analysis/analyzer.test.ts:126]: should it read more like a sentence?
```suggestion
  it('should not analyze Vue.js with type checks', async () => {
```

File: packages/jsts/tests/analysis/analyzer.test.ts [hunk 2]
```diff
@@ -133,14 +133,11 @@ describe('analyzeJSTS', () => {
...
-    } = analyzeJSTS(await jsTsInput({ filePath, tsConfigs }), language) as JsTsAnalysisOutput;
-    expect(issue1).toEqual(
-      expect.objectContaining({
-        ruleId: 'strings-comparison',
-      }),
-    );
+    const { issues } = analyzeJSTS(
+      await jsTsInput({ filePath, tsConfigs }),
+      language,
+    ) as JsTsAnalysisOutput;
```
Comment (zglicz) [inline packages/jsts/tests/analysis/analyzer.test.ts:139]: is the "as JsTsAnalysisOutput" necessary? Isn't that the exact return of the method?
```suggestion
    );
```

**Review by zglicz — COMMENTED** [review]
Just two suggestions from my side. I will leave it to approval from someone more knowledgeable.

File: its/plugin/sonarlint-tests/src/test/java/SonarLintTest.java [hunk 3]
```diff
@@ -175,9 +175,7 @@ void should_analyze_js_with_typed_rules() throws IOException {
     fileName = "file.vue";
     content = Files.readString(TestUtils.projectDir("js-sonarlint-project").resolve(fileName));
     issues = analyze(fileName, content);
-    assertThat(issues)
-      .extracting(Issue::getRuleKey)
-      .contains("javascript:S2870", "javascript:S3504");
+    assertThat(issues).extracting(Issue::getRuleKey).contains("javascript:S3504");
```
Comment (saberduck) [inline its/plugin/sonarlint-tests/src/test/java/SonarLintTest.java:178]: can we use `containsExactly`?

File: packages/jsts/src/builders/build.ts [hunk 4]
```diff
@@ -41,11 +41,13 @@ export function buildSourceCode(input: JsTsAnalysisInput, language: JsTsLanguage
       // enable logs for @typescript-eslint
       // debugLevel: true,
       filePath: input.filePath,
-      programs: input.programId && [getProgramById(input.programId)],
-      project: input.tsConfigs,
       parser: vueFile ? parsers.typescript.parser : undefined,
     };
     const parser = vueFile ? parsers.vuejs : parsers.typescript;
+    if (getContext()?.sonarlint || !vueFile) {
```
Comment (saberduck) [inline packages/jsts/src/builders/build.ts:47]: why we need optional chaining to access `getContext()`? shouldn't it be always available? I would prefer to have such critical code with as few conditions as possible, it makes debugging easier

File: packages/jsts/tests/analysis/analyzer.test.ts [hunk 5]
```diff
@@ -123,12 +123,18 @@ describe('analyzeJSTS', () => {
     );
   });
 
-  it('should analyze Vue.js code with type checks', async () => {
+  it('should analyze Vue.js code with type checks in SL', async () => {
```
Comment (saberduck) [inline packages/jsts/tests/analysis/analyzer.test.ts:126]: we should be consistent and have same behavior in SL and SQ, otherwise users will come back complaining

File: sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/bridge/AnalysisWithProgram.java [hunk 6]
```diff
@@ -178,7 +178,7 @@ private void analyze(InputFile file, @Nullable TsProgram tsProgram) throws IOExc
           file
         );
       } catch (IOException e) {
-        LOG.error("Failed to get response while analyzing " + file, e);
+        LOG.error("Failed to get response while analyzing " + file.uri(), e);
```
Comment (saberduck) [inline sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/bridge/AnalysisWithProgram.java:181]: why is this needed? should be the same

Comment (vdiez) [reply sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/bridge/AnalysisWithProgram.java:181]: the difference is this:

```
Expecting ArrayList:
 <["Failed to get response while analyzing dir/file.js",
    "Failure during analysis"]>
to contain:
 <["Failed to get response while analyzing file:///C:/Users/victor.diez/AppData/Local/Temp/junit11831790534385308678/dir/file.js"]>
```

I changed it to have the same output as `AnalysisWithWatchProgram.java`. Changing the type of program made the logs have different output, and thus the unit tests checking output failed because of that

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM!
